apis/container/v1beta1/zz_nodepool_types.go

```diff
@@ -715,6 +725,11 @@ type NodePoolNodeConfigParameters_2 struct {
 	// +kubebuilder:validation:Optional
 	ImageType *string `json:"imageType,omitempty" tf:"image_type,omitempty"`
 
+	// This is an injected field with a default value for being able to merge items of the parent object list.
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:="0"
+	Index *string `json:"index" tf:"-"`
```

Comment:
@JonathanO @ulucinar do you understand why is this the only of the 3 index fields that doesn't have `omitempty`?

Generating yaml of NodePool with NodePoolNodeConfigParameters_2  (with index left to zero value) produces `index: null`